default-cluster-templates/baseline-k3s.json

```diff
@@ -16,65 +16,65 @@
-                "--tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384"
-              ]
-            },
-            "kubeControllerManager": {
-              "extraArgs": null
-            },
-            "kubeScheduler": {
-              "extraArgs": null
-            },
-            "etcd": {
-              "exposeMetrics": false,
-              "backupConfig": {
-                "retention": "5",
-                "scheduleCron": "0 */5 * * *"
-              },
-              "customConfig": {
+          "kthreesConfigSpec": {
+            "files": [
+              {}
+            ],
+            "agentConfig": {
+              "kubelet": {
                 "extraArgs": [
-                  "cipher-suites=[TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384]"
+                  "--topology-manager-policy=best-effort",
+                  "--cpu-manager-policy=static",
+                  "--reserved-cpus=1",
+                  "--max-pods=250",
+                  "--tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384"
                 ]
               }
-            }
+            },
+            "preK3sCommands": [
+              "mkdir -p /etc/systemd/system/k3s-server.service.d",
+              "echo '[Service]\nEnvironmentFile=/etc/environment' > /etc/systemd/system/k3s-server.service.d/override.conf",
+              "mkdir -p /var/lib/kubelet/static-pods",
+              "sed -i '/kubelet-arg:/a\\- pod-manifest-path=/var/lib/kubelet/static-pods' /etc/rancher/k3s/config.yaml",
+              "sudo cp /var/lib/rancher/k3s/agent/pod-manifests/connect-agent.yaml /var/lib/kubelet/static-pods/connect-agent.yaml"
+            ],
+            "postK3sCommands": [
+              "kubectl apply -f https://raw.githubusercontent.com/projectcalico/calico/v3.29.3/manifests/calico.yaml",
+              "sudo mount --make-shared /sys && sudo mount --make-shared / && sudo mount --make-shared /run"
+            ],
+            "privateRegistriesConfig": {},
+            "serverConfig": {
+              "cni": "calico",
+              "cniMultusEnable": true,
+              "disableComponents": [],
+              "kubeApiServer": {
+                "extraArgs": [
+                  "--feature-gates=PortForwardWebsockets=true",
+                  "--tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384"
+                ]
+              },
+              "kubeControllerManager": {
+                "extraArgs": null
+              },
+              "kubeScheduler": {
+                "extraArgs": null
+              },
```

Comment:
kubeapi server, kubecontrollermanager and kubescheduler in rke2 has type 
`bootstrapv1alpha1.ComponentConfig`
[source](https://github.com/rancher/cluster-api-provider-rke2/blob/main/controlplane/api/v1alpha1/rke2controlplane_types.go#L137) 

when k3s has just `[]string` and a bit different name `KubeAPIServerArgs` [source](https://github.com/k3s-io/cluster-api-k3s/blob/main/bootstrap/api/v1beta2/kthreesconfig_types.go#L62)

so i believe the json "syntax" for k3s is bit different

after cluster creation can you check `ps aux |grep kube-apiserver` on edgenode of all extra args are passed correctly?

Reply:
yeah, that again need to be fixed in the template. Thanks to point out! 

As mentioned above, will try to minimize as possible. I noted that the tls-cipher-suites default value is the same as described in the template. Also, it seems that `feature-gates=PortForwardWebsockets=true`  became default when graduate to Beta in [v1.31](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates-for-alpha-or-beta-features)

btw, the result on the ps aux was nil

Reply:
another view over the ciphers, they are not the same. So, I will keep them the same